src/internet_identity/src/storage.rs

```diff
@@ -396,6 +425,79 @@ impl<M: Memory + Clone> Storage<M> {
         }
     }
 
+    #[allow(dead_code)]
+    pub fn from_memory_v6_to_v7(memory: M) -> Option<Self> {
+        let maybe_storage_v6 = Self::from_memory(memory.clone());
+        maybe_storage_v6.as_ref()?;
+        let storage_v6 = maybe_storage_v6.unwrap();
```

Comment:
It would be nicer to return `None` on empty rather than panicking, no?

Reply:
It will not panic, due to the check on the previous line.
But I noticed now that both lines can be combined, so I did that, and hopefully it is more clear now. Also, I added a test for the case when `maybe_storage_v6` is `None`.